pandas/core/internals/construction.py

```diff
@@ -475,23 +473,11 @@ def dict_to_mgr(
         keys = list(data.keys())
         columns = Index(keys)
         arrays = [com.maybe_iterable_to_list(data[k]) for k in keys]
-        # GH#24096 need copy to be deep for datetime64tz case
-        # TODO: See if we can avoid these copies
         arrays = [arr if not isinstance(arr, Index) else arr._data for arr in arrays]
-        arrays = [
-            arr if not is_datetime64tz_dtype(arr) else arr.copy() for arr in arrays
-        ]
 
     if copy:
-        # arrays_to_mgr (via form_blocks) won't make copies for EAs
         # dtype attr check to exclude EADtype-castable strs
-        arrays = [
-            x
-            if not hasattr(x, "dtype") or not isinstance(x.dtype, ExtensionDtype)
-            else x.copy()
-            for x in arrays
-        ]
-        # TODO: can we get rid of the dt64tz special case above?
+        arrays = [x if not hasattr(x, "dtype") else x.copy() for x in arrays]
```

Comment:
If I understand correctly, this will now copy all arrays that were present in the dict (if `copy=True`), instead of only EAdtype arrays. This ensures that we now honor the copy keyword for AM (i.e. the actual fix in the PR). 
But doesn't that also mean that this causes an extra copy for BM for numpy arrays?

Reply:
good catch, will update